### PR TITLE
Basic support for v1.Pod

### LINKED ISSUE #99: Support v1.Pod
It's another Kubernetes object kind.  It's a bit handy to be able to interact with it.  The missing support is mostly just the in-memory testing stuff.

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #99

### CODE REVIEW

File: src/txkube/testing/integration.py [hunk 1]
```diff
@@ -481,6 +487,74 @@ def check_configmaps(collection):
...
+        )
+
+
+    @async
+    @needs(namespace=creatable_namespaces())
+    def test_duplicate_pod_rejected(self, namespace):
+        """
+        ``IKubernetesClient.create`` returns a ``Deferred`` that fails with
+        ``KubernetesError`` if it is called with a ``Pod`` object with the
+        same name as a *Pod* which already exists in the same namespace.
```
Comment (markrwilliams) [inline src/txkube/testing/integration.py:510]: Why `Pod` (monospace) then _Pod_ (italics)?

File: src/txkube/testing/integration.py [hunk 2]
```diff
@@ -481,6 +487,74 @@ def check_configmaps(collection):
...
 
+class _PodTestsMixin(object):
+    @async
+    @needs(namespace=creatable_namespaces())
+    def test_pod(self, namespace):
+        """
+        ``Pod`` objects can be created and retrieved using the ``create`` and
+        ``list`` methods of ``IKubernetesClient``.
+        """
+        return self._create_list_test(
```
Comment (markrwilliams) [inline src/txkube/testing/integration.py:498]: `_create_list_test` isn't part of this PR, but that calls [`example()` on the strategy]( https://github.com/LeastAuthority/txkube/blob/cbaedf9dd3307b617edf6d1000161235fc9716b3/src/txkube/testing/integration.py#L869), which is `pods()` here.  You might consider changing that to use [`data.draw`](http://hypothesis.readthedocs.io/en/latest/data.html#drawing-interactively-in-tests), which in turn require that this be decorated with `given`.  Is the concern that doing so will run the test method more than once?

File: src/txkube/testing/strategies.py [hunk 3]
```diff
@@ -252,6 +252,16 @@ def podspecs():
     """
     return builds(
         v1.PodSpec,
+        activeDeadlineSeconds=one_of(
+            none(),
+            integers(min_value=0, max_value=2 ** 63 - 1),
```
Comment (markrwilliams) [inline src/txkube/testing/strategies.py]: Should this not be 2 ** 64 - 1?  Or is this a signed integer?  And if so, why is its minimum value 0?

I admit that having a negative deadline makes no sense, but then it also makes no sense to a use a signed integer.

**Review by markrwilliams — APPROVED** [review]
Minor comments and questions.  This is a pretty straight-forward PR.

File: src/txkube/testing/integration.py (hunk 2 above)
Comment (exarkun) [reply src/txkube/testing/integration.py:498]: It's primarily just the general problem of Deferred-returning tests being incompatible with `given`.

File: src/txkube/testing/integration.py (hunk 1 above)
Comment (exarkun) [reply src/txkube/testing/integration.py:510]: I guess I'm trying to differentiate between the Python object and the Kubernetes object.  Unsuccessfully, clearly. :)

File: src/txkube/testing/strategies.py (hunk 3 above)
Comment (exarkun) [reply src/txkube/testing/strategies.py]: Yep, it makes no sense, but it's how Kubernetes works.  This value is a 64 bit signed integer the minimum acceptable value of which is 0...  I'll add a comment and maybe a link to the spec.

**Pull request merged**